vscode-sbcl.md

```diff
@@ -0,0 +1,190 @@
+   Palette` and choose `Alive: Inline Eval`
+3. You will see a small pop up that says `=> 4 (3 bits, #x4, #o4,
+   #b100)`, which is the result
+
+   _Evaluating a statement in-line is exactly the same as sending it to
+   the REPL.  The only difference is how it is displayed._
+
+#### Evaluate a statement
+
+1. In your open file in your editor window, enter:
+   ~~~lisp
+   (+ 2 2)
+   ~~~
+2. Open the Command Palette on the menu at the top `View/Command
+   Palette` and choose `Alive: Send To REPL`
+3. You will see the expression show up in the REPL along with the result.
+   ~~~lisp
+   CL-USER>
+   (+ 2 2)
+   4
+   CL-USER>
+   ~~~
+
+#### Use the Interactive Debugger to abort
+
+1. In your open file in your editor window, enter:
+   ~~~lisp
+   (defun divide (x y)
+     (/ x y))
+   ~~~
+2. Put your cursor after the last parenthesis if it isn't already there.
+   Open the Command Palette on the menu at the top `View/Command
+   Palette` and choose `Alive: Inline Eval` to load your `define`
+   function into your image.
+3. In your open file, add another new line and enter:
+   ~~~lisp
+   (divide 1 0)
+   ~~~
+4. Put your cursor after the last parenthesis if it isn't already there.
+   Open the Command Palette on the menu at the top `View/Command
+   Palette` and choose `Alive: Inline Eval` to run your divide function
+   in your image.
+5. You will see the Interactive Debugger pop up.  In the `Restarts`
+   section, choose option 2 to Abort.
+6. You're now back to your editor and still-running REPL and can
+   continue like it never happened.
+
+#### Use the Interactive Debugger to fix the problem at runtime
+
+1. In your open file in your editor window, enter:
+   ~~~lisp
+   (defun divide (x y)
+     (assert (not (zerop y))
+	          (y)
+	          "The second argument can not be zero. Please change it.")
+     (/ x y))
+   ~~~
+2. Put your cursor after the last parenthesis if it isn't already there.
+   Open the Command Palette on the menu at the top `View/Command
+   Palette` and choose `Alive: Inline Eval` to load your `define`
+   function into your image.
+3. In your open file, add another new line and enter:
+   ~~~lisp
+   (divide 1 0)
+   ~~~
+4. Put your cursor after the last parenthesis if it isn't already there.
+   Open the Command Palette on the menu at the top `View/Command
+   Palette` and choose `Alive: Inline Eval` to run your divide function
+   in your image.
+5. You will see the Interactive Debugger pop up.  In the `Restarts`
+   section, choose option 0 to "Retry assertion with new value for Y".
+6. In the popup menu, enter `y`
+7. In the next popup menu, enter `1`
+8. You should now see a small pop up that says `=> 1 (1 bit, #x1, #o1,
+   #b1)`, which is the result of the new value.  You're now back to your
+   editor and still-running REPL after crashing out into the debugger,
+   having it let you change the value that caused the crash, and then
+   proceeding like you never typed that bad `0` value.
```

Comment:
Very cool. If you add a screenshot it's first class (it might help readers yeah?).

Reply:
I will add some screenshots and gifs asap.

Reply:
Done